File: Task41/app/src/main/java/edu/deakin/s600152989/sit305/task41/Task.java
package edu.deakin.s600152989.sit305.task41;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "task_table")
public class Task {

    @PrimaryKey(autoGenerate = true)
    private int id;

    private String title;
    private String description;
    private String startDate;
    private String dueDate;

    // Constructor
    public Task(String title, String description, String startDate, String dueDate) {
        this.title = title;
        this.description = description;
        this.startDate = startDate;
        this.dueDate = dueDate;
    }

    // Setter for ID (Room will auto-generate it)
    public void setId(int id) {
        this.id = id;
    }

    // Getters
    public int getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public String getDescription() {
        return description;
    }

    public String getStartDate() {
        return startDate;
    }

    public String getDueDate() {
        return dueDate;
    }
}

File: Task41/app/src/main/java/edu/deakin/s600152989/sit305/task41/Task.kt
package edu.deakin.s600152989.sit305.task41

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "task_table")
class Task // Constructor
    (@JvmField val title: String, @JvmField val description: String, @JvmField val startDate: String, @JvmField val dueDate: String) {
    // Setter for ID (Room will auto-generate it)
    // Getters
    @JvmField
    @PrimaryKey(autoGenerate = true)
    var id: Int = 0
}

